### PR TITLE
GTEST/UCT: Run tests for TCP over fastest device only in Jenkins

### PR DESCRIPTION
**Pull request by dmitrygladkov**
## What

Uses only the fastest device for UCT/TCP testing in Jenkins 

## Why ?

Reduce total testing time

## How ?

Implemented GTEST_UCT_TCP_DEVS variable that takes "all", "fastest", "", or a name of IP interface.
Use this variable in test_jenkins.sh to use the fastest device.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9647/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9648/ for details (Mellanox internal link).

File: test/gtest/uct/uct_test.cc [hunk 1]
```diff
@@ -183,6 +187,18 @@ std::vector<const resource*> uct_test::enum_resources(const std::string& tl_name
             ASSERT_UCS_OK(status);
 
             for (unsigned j = 0; j < num_tl_resources; ++j) {
+                if ((tl_resources[j].tl_name == static_cast<std::string>("tcp")) &&
```
Comment (evgeny-leksikov) [inline test/gtest/uct/uct_test.cc]: (std::string("tcp") == tl_resources[j].tl_name)

File: test/gtest/uct/uct_test.cc [hunk 2]
```diff
@@ -183,6 +187,18 @@ std::vector<const resource*> uct_test::enum_resources(const std::string& tl_name
...
+                    (tcp_devs != "all")) {
+                    if (tcp_dev_set) {
+                        continue;
+                    } else {
+                        if ((tcp_devs == "") || (tcp_devs == "fastest") ||
```
Comment (evgeny-leksikov) [inline test/gtest/uct/uct_test.cc]: How does it work? Is `tl_resources` sorted by speed?

File: test/gtest/uct/uct_test.cc [hunk 3]
```diff
@@ -147,6 +147,10 @@ void uct_test::set_sockaddr_resources(uct_md_h md, char *md_name, cpu_set_t loca
 
 std::vector<const resource*> uct_test::enum_resources(const std::string& tl_name,
                                                       bool loopback) {
+    // Run the tests for all devices for TCP by default
+    static char *env = getenv("GTEST_UCT_TCP_DEVS");
```
Comment (evgeny-leksikov) [inline test/gtest/uct/uct_test.cc]: according to name it might be a list of devices but it's handled like specific value

Comment (dmitrygladkov) [reply test/gtest/uct/uct_test.cc]: fixed

File: test/gtest/uct/uct_test.cc (hunk 1 above)
Comment (dmitrygladkov) [reply test/gtest/uct/uct_test.cc]: done

File: test/gtest/uct/uct_test.cc (hunk 2 above)
Comment (dmitrygladkov) [reply test/gtest/uct/uct_test.cc]: Yes, TCP provides a list of devvice (IP interfaces) sorted by their speeds. We rely on this fact, added the comment

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9651/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9653/ for details (Mellanox internal link).

**Pull request closed without merging**